### PR TITLE
doks: Produce a valid kubeconfig file.

### PR DESCRIPTION
**Pull request by andrewsomething**
As a side-effect of moving to the `GetCredentials` endpoint in https://github.com/terraform-providers/terraform-provider-digitalocean/pull/311, the `raw_config` is no longer a usable kubeconfig file. This PR attempts to fix that.

@snormore, I'd love your input here. This feels more fragile than I'd like. Is there a better way to get the cluster and user names for use in a kubeconfig file short of making calls to both `GetKubeConfig` and `GetCredentials`?

### CODE REVIEW

**Comment by snormore** [comment]
I think the way you're doing it here is the best approach we have. It's unfortunate that `GetCredentials` doesn't include a `kubeconfig` sub-field containing this already. One reason we wouldn't want to call both `GetCredentials` and `GetKubeconfig` is that we'd generate 2 tokens, one for each. The approach you're taking here seems reasonable. The values of the cluster name and user names can pretty much be anything as long as they're consistent through the kubeconfig.

File: digitalocean/resource_digitalocean_kubernetes_cluster_test.go [hunk 1]
```diff
@@ -372,3 +374,44 @@ func Test_filterTags(t *testing.T) {
...
+contexts:
+- context:
+    cluster: do-lon1-test-cluster
+    user: do-lon1-test-cluster-admin
+  name: do-lon1-test-cluster
+current-context: do-lon1-test-cluster
+users:
+- name: do-lon1-test-cluster-admin
+  user:
+    client-key-data: ""
```
Comment (snormore) [inline digitalocean/resource_digitalocean_kubernetes_cluster_test.go]: How about making `client-key-data` and `client-certificate-data` `omitempty`?

File: go.mod [hunk 2]
```diff
@@ -12,3 +12,5 @@ require (
 )
 
 replace git.apache.org/thrift.git => github.com/apache/thrift v0.0.0-20180902110319-2566ecd5d999
+
+go 1.13
```
Comment (snormore) [inline go.mod]: This kept getting re-added for me too, unsure why 🤔

Comment (andrewsomething) [reply go.mod]: Reverted.

**Pull request merged**